Sources/Quick/DSL/DSLValidation.swift

```diff
@@ -0,0 +1,28 @@
+/**
+    Descriptors intended to identify DSL function context.
+*/
+public enum ExampleContextDescriptor: String {
```

Comment:
Is this a reason for this to be public? I think we can leave this to `internal`. This project prefers being explicit about visibility scopes, so just replace `public` with `internal`.